mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +724,18 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def print_model_dependencies(model_uri):
```

Comment:
As discussed in https://github.com/mlflow/mlflow/pull/5503#issuecomment-1070513180, printing the dependencies isn't the end goal for most users (apologies for naming the task this way). Instead, users want to be able to install the dependencies in their current environment.

Can we adjust this method to instead download the dependencies to the local filesystem and return a path to the downloaded dependencies? This should be as simple as downloading `requirements.txt` and / or `conda.yaml` (depending on what the user wants).

We should let the user specify the dependencies format (e.g. `format="conda"`, `format="pip"`) that we can use to determine whether we resolve `requirements.txt` or `conda.yaml`. This should default to `format="pip"`.